yamllint: Disable `document-start` by default

adrienverge opened the pull request:
Some projects use documents starts (`---`) in YAML documents (for example Ansible, OpenStack, yamllint), others don't. Since these markers are required when declaring multiple documents in a single .yaml file (see [the spec](http://yaml.org/spec/1.2/spec.html#id2800132)), it is a bad idea to forbid them.

This commit disables the `document-start` rule by default, instead of forcing / forbidding the use of these markers.

Closes: #1417
Fixes: #923 #965

Makman2 commented on bears/yaml/YAMLLintBear.py:
```diff
@@ -33,13 +33,12 @@ def generate_config(filename, file,
-            yamllint_configs['rules']['document-start']['present'] = True
+        if document_start is not None:
+            yamllint_configs['rules']['document-start'] = {
+                'present': document_start}
```
For the sake of readability I think this is better:

```
    ...
    'rules': {}
    ...

# Don't forget to wrap the line to meet the 80 chars limit :D
yamllint_configs['rules']['document-start'] = 'disable' if document_start is None else {'present': document_start}
```

Instead of patching the dict depending on a value, we directly assign the correct value :)

Makman2 commented:
Commit message:

When you want to make use of auto-close features of GitHub, don't place a colon:
```
Closes xxxx
instead of
Closes: xxxx
```
also the "Fixed" issues are already closed, and this commit does not really "fix" them. You could write `Related to xxxx` if you want to refer those :)

Makman2 commented:
Also if you want to work on an issue @adrienverge, tell us so we can assign you and nobody's doing double-work :)
(or come by at gitter and assign yourself with `cobot assign <link-to-issue>` :D)

adrienverge commented:
> For the sake of readability I think this is better:

You're totally right, thanks! Updated.

> When you want to make use of auto-close features of GitHub, don't place a colon:

GitHub works fine with colons, but I removed them if you prefer this way.

> also the "Fixed" issues are already closed

I see a difference between "fixed" and "closed". My commit closes a discussion, but really fixes 1) the problem described in #923, 2) the other problem brought by commit in #965.

Makman2 commented:
> GitHub works fine with colons, but I removed them if you prefer this way.

Oh really? Sorry then :3

> I see a difference between "fixed" and "closed". My commit closes a discussion, but really fixes 1) the problem described in #923, 2) the other problem brought by commit in #965.

I think this is thought 3 corners around. Though those issues have not really identified the overall problem (by making the force-markers setting optional), they addressed a more specific issue which was itself solved by those other commits.

Also we use "Fixes" specifically for fixing bugs (we actually track bug statistics via commits), and "Closes" for every other issue we want to close. Just you know :)

Don't want to be nitpicking, but for those "Fixes xxx" issues I would insist that you change them to "Related to xxx" ^^

adrienverge commented:
> Don't want to be nitpicking, but for those "Fixes xxx" issues I would insist that you change them to "Related to xxx" ^^

Alright, done.

Makman2 commented:
ack 064abf4

Makman2 commented:
@rultor merge

rultor commented:
> @rultor merge

@makman2 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/10898-278434309)

rultor commented:
> @rultor merge

@makman2 Done! FYI, the full log is [here](http://www.rultor.com/t/10898-278434309) (took me 1min)
